### DISCUSSION TITLE
UI Suggestion - Updated top bar.

### DISCUSSION THREAD

**Discussion by ananas-wonders** [koreader/koreader#9188] [Ideas]
Hello all, 
Hope this is the right place to suggest this. 

Been using Koreader for a month now on kindle PW4. great experience. I would like to say thanks to all the developers who work on this project. 

I've been struggling with how much space the top menus take when open . I also get lost a lot reading all the menu items without visual cues. 

I thought I'd make a quick mock-up (using Adobe XD) for a refreshed top bar for a couple of menus as a proof of concept. 

#### Style Menu: 
![Menu 1](https://user-images.githubusercontent.com/7282452/172864309-b6728ec1-ce3b-4695-8a9d-107daa30d25e.png)

#### Book Menu:
![Menu 2](https://user-images.githubusercontent.com/7282452/172864340-ac74beec-311c-4932-9906-b5ac3d80644c.png)

I'm not a developer, so I honestly don't know if it's possible to have menus like this (from a device resources stand point): 

- Having that many icons (Even if svg/xml) would it need more processing power?
- Having a slider menu, would it look good on E-ink displays? 

And the last question is... are the developers open to this suggestion to begin with :) ? (sorry if I'm intruding). 

Have a good day.

**Comment by poire-z** [comment]
> are the developers open to this suggestion to begin with :) ? (sorry if I'm intruding).

We are. (You're not :)

> Having that many icons (Even if svg/xml) would it need more processing power?
> Having a slider menu, would it look good on E-ink displays?

None should be an issue.

> I honestly don't know if it's possible to have menus like this (from a device resources stand point)

Device resources are ok - but developer resources may be less :)

> I thought I'd make a quick mock-up

You're choice of icons is quite nice and expressive.

But: :)

> I've been struggling with how much space the top menus take when open . I also get lost a lot reading all the menu items without visual cues.

You shouldn't spend that much time reading the top menu :) Your book below the menu is much more interesting :)

Anyway, a few (some personal) thoughts about why our current stacked text menu feels fine/better:
- the first level menus are verbose and just a bunch of various items : they _can_ take space over the book content, they are the entry points to other things. Icons could/would look nice, but a user would just be passing by to get to other things :)
- for 2nd++ levels menu, that may have effects on the book content below, we try to keep them small (ie. 5 font names, 5/6 style tweaks) to leave enough book content so one can see how the changes apply on the content
- our menu items need to be translatable - and some languages need a bit more wording than what English allows (English allows you to have small text under your icons, with other languages we may need unclear abbreviations). Also, "Style tweaks" is clearer than just "Tweaks" under an icon.
- some of our menu items give more info (ie. `Typography: English`, check or unchecked, some numbers) that would be lost with just icons (or complicated to draw, like your "3" on the Tweaks icons)
- we have separators/lines - so in the screenshots of our current menu, we know that `Settings` are related to the 2 other items in its section (Table of contents, Bookmarks)
- it's currently quite easy to add a menu item with text - if going with icons, each new menu item will need more thoughts to find it a nice icon, and a nice icon consistent with the others (which might be harder and need more discussions and time than implementing the feature :)
- it would be quite a lot of work on the TouchMenu code, with possible different looks for top level and below levels, or levels where text is prefered (ie. Fonts) and those with icons
- having something generic and consistent text-based feels preferable to me, than having a combinations of types (even if your screenshots and icons are quite pretty).

**Comment by mergen3107** [comment]
@ananas-wonders 
Icons look really neat! Almost conforming to the bottom menu icons.
I agree that a lot of space is wasted in the top menu.

`Having a slider menu, would it look good on E-ink displays?`
How about making pages of icons? Without scroll, just static redraws.


@poire-z 
`some of our menu items give more info ` : how about a combination of icons + text then? Maybe some obvious things can be represented as icons only, some of them as icon + text. Alternatively, make some icons non-square to fit text. 

(as a side note, if we have `Typography: English` should we also have, e.g., font name next to `Font`, style name next to `Style`, etc? I mean I can't remember how many times I checked font name by going into the font menu. It would be really handy if that info is provided as soon as I open the top menu)

Another thought is that those whole develop plugins should also be taking care of the icon. Or should this be centralized like release illustrations?

**Comment by mergen3107** [comment]
@poire-z 
Remember when we had debates about which items go to which tab, when we were reorganizing the bottom menu? With these new icons I have an idea. Can we use icons and their shade to show on/off status (like, shaded icon background ON, white one - OFF. Or insets like in PC UI buttons when they are pressed)? For example, `Sync T/B Margins` does not need so much horizontal space.

Then we can combine those single-icon options (along with moving some items like embedded fonts and styles to the top menu #9190 ) which will give us some room to re-arrange the bottom menu and make it more grouped (like those options about reflow are not scattered all around the tabs).

**Comment by poire-z** [comment]
> should we also have, e.g., font name next to Font, style name next to Style, etc? I mean I can't remember how many times I checked font name by going into the font menu

That must have been in your early years :) We have that (for Font:) since a year ago: #7490 (not sure why it does not appear in the first screenshot above - when the 2nd has Page map which is more recent).

> how about a combination of icons + text then? Maybe some obvious things can be represented as icons only, some of them as icon + text. Alternatively, make some icons non-square to fit text.
Another thought is that those whole develop plugins should also be taking care of the icon. Or should this be centralized like release illustrations?

Consistency, pain of adding/maintaining icons vs. necessity :)

As for the bottom menu suggestions, icons would probably be more welcome here (bottom menu for PDF has a bit more of them) - but the bottom menu code is bigger and messier and scarier and fragiler than any other bit of code, that I don't think anybody would have fun working on it - so any change from name on the left, widget on the right - like making a grid, multiple different widget/setting per line - is hard.

**Comment by mergen3107** [comment]
`We have that (for Font:) since a year ago:` 
Ohh, my apologies! It does in fact show up, I don't know why the memory of the name not being there stuck so strongly. Maybe because I am reading epubs with embedded fonts for a long time and didn't actually get to choose my own font xD

`but the bottom menu code is bigger and messier and scarier and fragiler than any other bit of code`
Oh, ok! 

Again, thank you for your time and explanation! I hope I am not bothering you guys too much with my hypothetical ideas :`)
